src/PrestaShopBundle/Security/Admin/AdminAuthenticationSuccessHandler.php

```diff
@@ -0,0 +1,99 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Security\Admin;
+
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShopBundle\Entity\Employee\Employee;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use PrestaShopBundle\Entity\Tab;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+/**
+ * This handle is called when the employee successfully logs in to the back office, its purpose is
+ * to dynamically set the route to redirect to based on the Employee's configured homepage.
+ */
+class AdminAuthenticationSuccessHandler implements AuthenticationSuccessHandlerInterface
+{
+    use TargetPathTrait;
+
+    public function __construct(
+        private readonly RouterInterface $router,
+        private readonly Security $security,
+        private readonly TabRepository $tabRepository,
+        private readonly LegacyContext $legacyContext,
+    ) {
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token): ?Response
+    {
+        if ($request->hasPreviousSession()) {
+            $redirectUrl = $this->getTargetPath($request->getSession(), 'main');
+        }
+        if (empty($redirectUrl)) {
+            $redirectUrl = $this->getHomepageUrl();
+        }
+        if (empty($redirectUrl)) {
+            $redirectUrl = $this->router->generate('admin_homepage');
+        }
+
+        return new RedirectResponse($redirectUrl);
+    }
+
+    private function getHomepageUrl(): ?string
+    {
+        $loggedUser = $this->security->getUser();
+        if ($loggedUser instanceof Employee) {
+            $homeUrl = null;
```

Comment:
Why don't initialize `$homeUrl` with `$this->legacyContext->getAdminLink('AdminDashboard');` ?

Reply:
The home url is not necessarily the `AdminDashboard`, you can configure your own default page in the Employee profile, that's why it's dynamic and based on the employee's configuration

As such a new dedicated Symfony controller was implemented `admin_homepage` which is bound to the `/` URL and handles this dynamic redirection, so actually here we could use this `admin_homepage` route which is better suited than the redirection to `AdminDashboard`

But if e redirect to a route that is also only handling a redirection it's an extra request that we can avoid by redirecting to the appropriate URL here right after the login

One improvement though could be to extract this homepage URL fetching process in a dedicated server because this class and the `homepageAction` controller do have common and duplicated code I'll add this small refacto in the TO LIST in the epic https://github.com/PrestaShop/PrestaShop/issues/36001